src/interactivity/win32/windowio.cpp

```diff
@@ -167,6 +167,14 @@ void HandleKeyEvent(const HWND hWnd,
         // --- END LOAD BEARING CODE ---
     }
 
+    // Certain applications like PowerToys and AutoHotKey, due to their keyboard
+    // remapping feature, send us key events using SendInput() whose values could
+    // be outside of the valid range. We should ignore such KeyEvents. GH#7064
+    if (VirtualKeyCode == 0 || VirtualKeyCode >= 0xff)
+    {
+        return;
+    }

```

Comment:
`HandleKeyEvent` is called for various message types (see line 149 above), but we should only discard these events for `WM_KEYUP` and `WM_KEYDOWN`, which is also what Windows Terminal does. I believe you can fix this issue if you merge this if condition with the above if condition as an `else if`.

Additionally, I'm wondering why this code doesn't test for `VirtualScanCode == 0` like WT does. Is there a reason for that?

Reply:
To be honest I'm not entirely sure either. The way I wrote the linked PR for WT is by testing a ton of different applications that use `SendMessage` and trying to see what would correctly end up ignoring such messages. But I don't think I've considered testing whether `VirtualKeyCode == 0 || VirtualKeyCode >= 0xff` alone is already enough to filter them out. It might just work!